test/index.test.js

```diff
@@ -88,77 +66,61 @@ describe('index', () => {
             };
-            breakRunMock.runCommand.resolves(fakeResponse);
+            expectedOptions = {
+                url: `${apiUrl}/mynewbranch?access_key=myAccessToken`,
+                method: 'GET'
+            };
+            requestMock.yieldsAsync(null, fakeResponse, fakeResponse.body);
         });
 
         it('resolves to the correct parsed url for ssh', () => {
             const expected =
                 'bitbucket.org:batman/{de7d7695-1196-46a1-b87d-371b7b2945ab}:master';
-            const expectedOptions = {
-                url: `${apiUrl}master`,
-                method: 'GET',
-                login_type: 'oauth2',
-                oauth_access_token: token
+
+            expectedOptions = {
+                url: `${apiUrl}/master?access_key=myAccessToken`,
+                method: 'GET'
             };
 
             return scm.parseUrl({
                 checkoutUrl: 'git@bitbucket.org:batman/test.git#master',
                 token
             }).then((parsed) => {
-                assert.calledWith(breakRunMock.runCommand, expectedOptions);
+                assert.calledWith(requestMock, expectedOptions);
                 assert.equal(parsed, expected);
             });
         });
 
         it('resolves to the correct parsed url for https', () => {
             const expected =
                 'bitbucket.org:batman/{de7d7695-1196-46a1-b87d-371b7b2945ab}:mynewbranch';
-            const expectedOptions = {
-                url: `${apiUrl}mynewbranch`,
-                method: 'GET',
-                login_type: 'oauth2',
-                oauth_access_token: 'myAccessToken'
-            };
 
             return scm.parseUrl({
                 checkoutUrl: 'https://batman@bitbucket.org/batman/test.git#mynewbranch',
                 token: 'myAccessToken'
             }).then((parsed) => {
-                assert.calledWith(breakRunMock.runCommand, expectedOptions);
+                assert.calledWith(requestMock, expectedOptions);
                 assert.equal(parsed, expected);
             });
         });
 
         it('rejects if request fails', () => {
             const err = new Error('Bitbucket API error');
-            const expectedOptions = {
-                url: `${apiUrl}mynewbranch`,
-                method: 'GET',
-                login_type: 'oauth2',
-                oauth_access_token: 'myAccessToken'
-            };
 
-            breakRunMock.runCommand.rejects(err);
+            requestMock.yieldsAsync(err);
 
             return scm.parseUrl({
                 checkoutUrl: 'https://batman@bitbucket.org/batman/test.git#mynewbranch',
                 token: 'myAccessToken'
             })
-            .then(() => assert.fail('Should not get here'))
-            .catch((error) => {
-                assert.calledWith(breakRunMock.runCommand, expectedOptions);
-                assert.deepEqual(error, err);
-            });
+                .then(() => assert.fail('Should not get here'))
```

Comment:
Note:

`.then()` takes 2 functions as arguments: the first one is invoked if the previous promise in the chain resolves and the second one is invoked if the previous promise in the chain rejects.  You could combine the `.catch()` statement into this `.then()` call so your test will error out with the correct message.